### PR TITLE
TSDK-540 DAML Module For Tetra

### PR DESCRIPTION
**Pull request by mundacho**
## Purpose

The goal of this PR is to create the contracts for the DAML integration of Dapps to the node.

## Approach

We did a cleanup of the contents of the repository and proceeded to implement four new modules.

- The Levels module
- The Onboarding module
- The Utils module
- The Wallet module

## Testing

We created DAML tests for the Wallet and Levels module.

## Tickets
* TSDK-540

### CODE REVIEW

File: daml/Topl/Levels.daml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      reason: Text
+        where
+          signatory requestor, operator
+          observer contractParticipants
+
+          key (operator, requestId) : (Party, Text)
+
+          maintainer key._1
+
+          choice LvlTransferRequest_Archive: ()
```
Comment (nandotorterolo) [inline daml/Topl/Levels.daml:36]: question: does the name of the choice should be the template name followed by _ : "LvlTransferRejected_Archive" ?

File: pom.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
     <groupId>co.topl</groupId>
     <artifactId>daml-bifrost-module</artifactId>
     <packaging>jar</packaging>
-    <version>0.2.0-SNAPSHOT</version>
+    <version>2.0.0-alpha1-SNAPSHOT</version>
```
Comment (nandotorterolo) [inline pom.xml:11]: maybe the description of the Pom, should be aligned with the readme,  just:
"`daml-bifrost-module` is a set of [DAML Modules](https://www.digitalasset.com/developers) for interfacing with the Topl blockchain and the DAML ledger."

File: daml/Topl/Wallet.daml [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+      currentPartyIndex: Int
+      currentContractIndex: Int
+        where
+          signatory owner, operator
+
+
+          key owner : Party
+          maintainer key
+
+          choice Vault_AcceptInvitation : ContractId WalletParty
```
Comment (nandotorterolo) [inline daml/Topl/Wallet.daml:36]: same question about names in choices

**Review by nandotorterolo — APPROVED** [review]
LGTM, (I am not familiar with daml logic behaviour. :( )

**Review by SeanCheatham — APPROVED** [review]

File: daml/Topl/Levels.daml (hunk 1 above)
Comment (mundacho) [reply daml/Topl/Levels.daml:36]: the names of the choices need to be unique. It is common to have an Archive choice, that is why we use the the contract name as prefix.

File: pom.xml (hunk 2 above)
Comment (mundacho) [reply pom.xml:11]: fixed

**Pull request merged**